### ISSUE TITLE
wrapped_units.LayerNormBasicLSTMNetwork

### ISSUE THREAD

**Issue by haoran8899** [tensorflow/models#1376]
No1.
http://stackoverflow.com/questions/43509103/typeerror-init-got-an-unexpected-keyword-argument-reuse

for No1,i modify the file dragnn/python/wrapped_units.py by del del the ", reuse=True" in 
"return tf.contrib.rnn.LayerNormBasicLSTMCell(
          num_units, layer_norm=self._attrs['layer_norm'], reuse=True)"  to slove the problem.

No2. using in dragnn model as blow:
 tagger.set_network_unit(name='wrapped_units.LayerNormBasicLSTMNetwork', hidden_layer_sizes='256)
when i run 
"text = '他們 是 親朋 好友 .'
tokens = [sentence_pb2.Token(word=word, start=-1, end=-1) for word in text.split()]
sentence = sentence_pb2.Sentence()
sentence.token.extend(tokens)
with tf.Session(graph=graph) as sess:
    # Restore the model we just trained.
    builder.saver.restore(sess, CHECKPOINT_FILENAME
)
    annotations, traces = sess.run([annotator['annotations'], annotator['traces']],
                      feed_dict={annotator['input_batch']: [sentence.SerializeToString()]})
HTML(visualization.trace_html(traces[0]))"

jupyter notebook show log as blow
[W 11:51:42.335 NotebookApp] Saving untrusted notebook trainer_tutorial.ipynb
I dragnn/core/compute_session_pool.cc:55] Destroying pool: total number of sessions created = 1
*** Error in `/usr/bin/python2.7': double free or corruption (!prev): 0x000000000109e160 ***
[I 11:52:25.327 NotebookApp] KernelRestarter: restarting kernel (1/5)
WARNING:root:kernel ad3ac582-fb41-4ffe-9a4f-e7b41ca1eabf restarted
 
source:
IN[1]: 
import os.path
import time
import random
import tensorflow as tf
from IPython.display import HTML
from tensorflow.python.platform import gfile
from tensorflow.python.platform import tf_logging as logging
from google.protobuf import text_format
from syntaxnet.ops import gen_parser_ops
from syntaxnet import load_parser_ops
from syntaxnet import task_spec_pb2
from syntaxnet import sentence_pb2
from dragnn.protos import spec_pb2 
from dragnn.python.sentence_io import ConllSentenceReader
from dragnn.python import evaluation
from dragnn.python import graph_builder
from dragnn.python import lexicon
from dragnn.python import load_dragnn_cc_impl
from dragnn.python import render_parse_tree_graphviz
from dragnn.python import render_spec_with_graphviz
from dragnn.python import spec_builder
from dragnn.python import trainer_lib
from dragnn.python import visualization

DATA_DIR = '/usr/cep/nlp/np/data/zht'
TENSORBOARD_DIR = '/usr/local/models/syntaxnet/tensorflow/tensorflow/tensorboard'
CHECKPOINT_FILENAME = '{}/zht.checkpoint'.format(DATA_DIR)
TRAINING_CORPUS_PATH = '{}/zh-ud-train.conllu'.format(DATA_DIR)
DEV_CORPUS_PATH = '{}/zh-ud-dev.conllu'.format(DATA_DIR)

assert os.path.isfile(TRAINING_CORPUS_PATH), '训练语料库不存在'

logging.set_verbosity(logging.WARN)

lexicon.build_lexicon(DATA_DIR, TRAINING_CORPUS_PATH)

lookahead = spec_builder.ComponentSpecBuilder('lookahead')
lookahead.set_network_unit(
    name='FeedForwardNetwork', hidden_layer_sizes='256')
lookahead.set_transition_system(name='shift-only', left_to_right='true')
lookahead.add_fixed_feature(name='words', fml='input.word', embedding_dim=32)
lookahead.add_rnn_link(embedding_dim=-1)
lookahead.fill_from_resources(DATA_DIR)

tagger = spec_builder.ComponentSpecBuilder('tagger')
tagger.set_network_unit(name='wrapped_units.LayerNormBasicLSTMNetwork', hidden_layer_sizes='256')
tagger.set_transition_system(name='tagger')
tagger.add_rnn_link(embedding_dim=-1)
tagger.add_token_link(source=lookahead, fml='input.focus', embedding_dim=32)
tagger.fill_from_resources(DATA_DIR)

parser = spec_builder.ComponentSpecBuilder('parser')
parser.set_network_unit(name='FeedForwardNetwork', hidden_layer_sizes='256',
                        layer_norm_hidden='true')
parser.set_transition_system(name='arc-standard')
parser.add_token_link(source=lookahead, fml='input.focus', embedding_dim=32)
parser.add_token_link(
    source=tagger, fml='input.focus stack.focus stack(1).focus',
    embedding_dim=32)

parser.add_fixed_feature(name='labels', embedding_dim=16,
                         fml=' '.join([
                             'stack.child(1).label',
                             'stack.child(1).sibling(-1).label',
                             'stack.child(-1).label',
                             'stack.child(-1).sibling(1).label',
                             'stack(1).child(1).label',
                             'stack(1).child(1).sibling(-1).label',
                             'stack(1).child(-1).label',
                             'stack(1).child(-1).sibling(1).label',
                             'stack.child(2).label',
                             'stack.child(-2).label',
                             'stack(1).child(2).label',
                             'stack(1).child(-2).label']))

parser.add_link(
        source=parser,  # recurrent connection
        name='rnn-stack',  # unique identifier
        fml='stack.focus stack(1).focus',  # look for both stack tokens
        source_translator='shift-reduce-step',  # maps token indices -> step
        embedding_dim=32)  # project down to 64 dims

parser.fill_from_resources(DATA_DIR)

master_spec = spec_pb2.MasterSpec()
master_spec.component.extend([lookahead.spec, tagger.spec, parser.spec])
HTML(render_spec_with_graphviz.master_spec_graph(master_spec))

IN[2]:
graph = tf.Graph()
with graph.as_default():
    hyperparam_config = spec_pb2.GridPoint(
        learning_method='adam',
        learning_rate=0.0005, 
        adam_beta1=0.9, adam_beta2=0.9, adam_eps=0.001,
        decay_steps=128000,
        dropout_rate=0.8, gradient_clip_norm=1,
        use_moving_average=True,
        seed=1)    
    builder = graph_builder.MasterBuilder(master_spec, hyperparam_config)
    target = spec_pb2.TrainTarget(
        name='all',
        unroll_using_oracle=[False, True, True], # train tagger & parser on gold unrolling, skip lookahead
        component_weights=[0, 0.5, 0.5]) # tagger and parser losses have equal weights
    trainer = builder.add_training_from_config(target)
    annotator = builder.add_annotation(enable_tracing=True)
    builder.add_saver()

N_STEPS = 20
BATCH_SIZE = 64
with tf.Session(graph=graph) as sess:
    sess.run(tf.global_variables_initializer())
    training_corpus = ConllSentenceReader(
        TRAINING_CORPUS_PATH, projectivize=True).corpus()
    dev_corpus = ConllSentenceReader(DEV_CORPUS_PATH).corpus()[:200]
    for step in xrange(N_STEPS):
        trainer_lib.run_training_step(sess, trainer, training_corpus, batch_size=BATCH_SIZE)
        tf.logging.warning('Step %d/%d', step + 1, N_STEPS)
    parsed_dev_corpus = trainer_lib.annotate_dataset(sess, annotator, dev_corpus)
    pos, uas, las = evaluation.calculate_parse_metrics(dev_corpus, parsed_dev_corpus)
    tf.logging.warning('POS %.2f UAS %.2f LAS %.2f', pos, uas, las)
    builder.saver.save(sess, CHECKPOINT_FILENAME)

IN[3]:
text = '他們 是 親朋 好友 .'
tokens = [sentence_pb2.Token(word=word, start=-1, end=-1) for word in text.split()]
sentence = sentence_pb2.Sentence()
sentence.token.extend(tokens)

with tf.Session(graph=graph) as sess:
    builder.saver.restore(sess, CHECKPOINT_FILENAME)
    annotations, traces = sess.run([annotator['annotations'], annotator['traces']],
                      feed_dict={annotator['input_batch']: [sentence.SerializeToString()]})
HTML(visualization.trace_html(traces[0]))

**Comment by drpngx** [comment]
Related to #9328, but this one might be related to loading op libraries.

**Comment by karmel** [comment]
@keveman can you comment?

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
